controller/src/beerocks/master/son_management.cpp

```diff
@@ -2110,10 +2110,99 @@ void son_management::handle_bml_message(Socket *sd,
             break;
         }
 
-        //TODO: add client to persistent DB if required and set client parameters
-        response->result() = 0; //Success.
+        auto send_response = [&](bool result) -> bool {
+            response->result() = (result) ? 0 : 1;
+            return message_com::send_cmdu(sd, cmdu_tx);
+        };
 
-        message_com::send_cmdu(sd, cmdu_tx);
+        // If none of the parameters is configured - return error.
+        // This flow should be blocked by the BML interface and should not be reached.
+        if ((request->client_config().stay_on_initial_radio == PARAMETER_NOT_CONFIGURED) &&
+            (request->client_config().stay_on_selected_device == PARAMETER_NOT_CONFIGURED) &&
+            (request->client_config().selected_bands == PARAMETER_NOT_CONFIGURED)) {
+            LOG(ERROR)
+                << "Received ACTION_BML_CLIENT_SET_CLIENT request without parameters to configure";
+            send_response(false);
+        }
+
+        // Get client mac.
+        auto client_mac = request->sta_mac();
+
+        // If client doesn't have node in runtime DB - add node to runtime DB.
+        if (!database.has_node(client_mac)) {
+            LOG(DEBUG) << "Setting a client which doesn't exist in DB, adding client to DB";
+            if (!database.add_node(client_mac)) {
+                LOG(ERROR) << "Failed to add client node for client " << client_mac;
+                send_response(false); //Fail.
```

Comment:
Ther comment is redundant, please remove it.

Reply:
removed the comment